prefect_kubernetes/worker.py

```diff
@@ -603,6 +610,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._created_secrets = {}
 
+    @retry(
+        stop=stop_after_attempt(MAX_ATTEMPTS),
+        wait=wait_fixed(RETRY_MIN_DELAY_SECONDS)
+        + wait_random(
+            RETRY_MIN_DELAY_JITTER_SECONDS,
+            RETRY_MAX_DELAY_JITTER_SECONDS,
+        ),
+        reraise=True,
+    )
```

Comment:
This should be added to `_create_job` instead. If we add it, here we'd resubmit jobs if a failure occurs while watching the job.